elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/ingest/Processor.scala

```diff
@@ -0,0 +1,41 @@
+package com.sksamuel.elastic4s.requests.ingest
+
+import com.sksamuel.elastic4s.{XContentBuilder, XContentFactory}
+
+/**
+  * Abstract representation of a processor with a constant name (e.g. "geoip") and an [[XContentBuilder]] that
+  * constructs the body (e.g. { "field" : "ip" } ).
+  */
+trait Processor {
+  def name: String
+  def buildProcessorBody(): XContentBuilder
```

Comment:
@sksamuel I followed the same structure as the `CustomQuery#buildQueryBody` method: https://github.com/sksamuel/elastic4s/blob/919875f4280a96cbf9307bb61a3386437a0cf33f/elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/searches/queries/Query.scala#L12